presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/executor/ConcurrentPhaseExecutor.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.benchmark.executor;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.benchmark.event.BenchmarkQueryEvent;
+import com.facebook.presto.benchmark.framework.BenchmarkQuery;
+import com.facebook.presto.benchmark.framework.BenchmarkQueryExecutor;
+import com.facebook.presto.benchmark.framework.BenchmarkRunnerConfig;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionService;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorCompletionService;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.facebook.presto.benchmark.event.BenchmarkQueryEvent.EventStatus;
+import static com.facebook.presto.benchmark.event.BenchmarkQueryEvent.EventStatus.FAILED;
+import static com.facebook.presto.benchmark.event.BenchmarkQueryEvent.EventStatus.SUCCEEDED;
+import static java.lang.Thread.currentThread;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+
+public class ConcurrentPhaseExecutor
+        implements PhaseExecutor
+{
+    private static final Logger log = Logger.get(ConcurrentPhaseExecutor.class);
+
+    private final int maxConcurrency;
+
+    private BenchmarkQueryExecutor benchmarkQueryExecutor;
+    private List<String> queries;
+    private Map<String, BenchmarkQuery> queryMap;
+    private Map<String, String> sessionProperties;
+
+    private ExecutorService executor;
+    private CompletionService<BenchmarkQueryEvent> completionService;
+    private AtomicInteger queriesSubmitted = new AtomicInteger();
```

Comment:
- You don't need `queriesSubmitted` to be a field - it can just be a local variable.
- You don't need it to be an `AtomicInteger` - it can just be an `int`.